api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -70,6 +76,25 @@ public static Type fromTypeName(String typeString) {
       return TYPES.get(lowerTypeString);
     }
 
+    if (lowerTypeString.startsWith("geometry")) {
+      Matcher geometry = GEOMETRY_PARAMETERS.matcher(typeString.substring(8));
+      if (geometry.matches()) {
+        return GeometryType.of(geometry.group(1));
+      }
+    }
+
+    if (lowerTypeString.startsWith("geography")) {
+      Matcher geography = GEOGRAPHY_PARAMETERS.matcher(typeString.substring(9));
+      if (geography.matches()) {
+        String algorithmName = geography.group(2);
+        EdgeAlgorithm algorithm =
+            (algorithmName == null || algorithmName.isEmpty())
```

Comment:
Why is `algorithmName.isEmpty` valid here? I think it should be an error if the type is passed as `"geography(srid:1234,)"`. If there is a non-null string then it must be a valid algorithm name. When there is no comma, the second capture group is null, so removing this `isEmpty` check will fix it.

Reply:
Removed `isEmpty`.